sklearn/metrics/tests/test_classification.py

```diff
@@ -197,6 +197,14 @@ def test_precision_recall_f_extra_labels():
         assert_raises(ValueError, recall_score, y_true_bin, y_pred_bin,
                       labels=np.arange(-1, 4), average=average)
 
+    # tests non-regression on issue #10307
+    y_true = np.array([[0, 1, 1], [1, 0, 0]])
+    y_pred = np.array([[1, 1, 1], [1, 0, 1]])
+    p, r, f, _ = precision_recall_fscore_support(y_true, y_pred,
+                                                 average='samples',
+                                                 labels=[0, 1])
+    assert_almost_equal(np.array([p, r, f]), np.array([3. / 4, 1., 5. / 6]))
```

Comment:
we have imported division from future on the top so let's remove the useless `.` for the float.